assignments/this.js

```diff
@@ -10,17 +17,54 @@
 */
 
 // Principle 1
+// ===========code example for Window Binding========
+// method
+// Create an object:
+const person = {
+    firstName: "John",
+    lastName : "Doe",
+    id     : 26,
+    fullName : function() {
+      return this.firstName + " " + this.lastName;
+    }
+  };
+
 
-// code example for Window Binding
 
 // Principle 2
+//========== code example for Implicit Binding======
+
+const MyObject = function (){
+    this.name = 'MyObjectName';
+    this.myProperty = 'property';
+  };
+ 
+  MyObject.prototype.doStuff = function (action) {
+    console.log(this.name + ' is ' + action + '!');
+  }
+ 
+  const obj = new MyObject();
+ 
+  obj.doStuff('awesome'); // prints 'MyObjectName is awesome!'
 
-// code example for Implicit Binding
 
 // Principle 3
+// ===============code example for New Binding============
+const sayMyName = function () {
+    console.log('My name is ' + this.name);
+  };
 
-// code example for New Binding
+  const jake = {
+    name: 'Jake'
+  }
 
-// Principle 4
+  const sayMyName = sayMyName.bind(jake);
```

Comment:
This is example of explicit binding. Example of "new" binding would be creating a class/blurprint for Object and making an instance of it by calling `new`, like `let Mike = new Person("Mike")`.